sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,319 @@
+                            attributeName, name, ForbidCertainMethodsCheck.class.getSimpleName()));
+                }
+            }
+            String argCountRegex = null;
+            if (Arrays.stream(childConf.getAttributeNames()).anyMatch(
+                item -> item.equals(AllowedKeyInConfig.METHOD_ARG_COUNT.keyName))) {
+                argCountRegex = childConf.getAttribute(AllowedKeyInConfig.METHOD_ARG_COUNT.keyName);
+            }
+            rules.add(new Rule(name,
+                    childConf.getAttribute(AllowedKeyInConfig.METHOD_NAME.keyName),
+                    argCountRegex,
+                    childConf.getAttribute(AllowedKeyInConfig.REASON.keyName)));
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.METHOD_CALL,
+            TokenTypes.LITERAL_NEW,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL: {
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                if (dot.getType() == TokenTypes.IDENT) {
+                    final String methodName = dot.getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+                // method that looks like: obj.method()
+                else {
+                    final String methodName = dot.getLastChild().getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+            }
+            // constructor
+            case TokenTypes.LITERAL_NEW: {
+                if (!includeConstructor) {
+                    break;
+                }
+                final DetailAST constructor = ast.getFirstChild();
+                // case for java8 expression: File::new
+                if (constructor == null) {
+                    break;
+                }
+                final String constructorName = constructor.getText();
+                final DetailAST arguments = ast.findFirstToken(TokenTypes.ELIST);
+                // case for expression: int int[4]
+                if (arguments == null) {
+                    break;
+                }
+                final int numberOfParameters = arguments.getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, constructorName, numberOfParameters);
+                break;
+            }
+            default:
+                break;
```

Comment:
`default` should throw an exception and we have a test util to do this. Look at our other checks.